### PR TITLE
[Data Platform] Add Data Catalog Policy tag

### PR DESCRIPTION
**Pull request by lcaggio**
Add Data Catalog Policy tag to `data-platform-foundations`.

### CODE REVIEW

File: examples/data-solutions/data-platform-foundations/06-common.tf [hunk 1]
```diff
@@ -21,27 +21,53 @@ module "common-project" {
...
     "datacatalog.googleapis.com",
     "dlp.googleapis.com",
   ])
 }
 
+# Data Catalog Policy tag
+
+module "common-datacatalog" {
+  count      = var.data_catalog_tags != null ? 1 : 0
+  source     = "../../../modules/data-catalog-policy-tag"
```
Comment (ludoo) [inline examples/data-solutions/data-platform-foundations/06-common.tf:64]: source always first :)

File: examples/data-solutions/data-platform-foundations/06-common.tf [hunk 2]
```diff
@@ -21,27 +21,53 @@ module "common-project" {
...
+  name       = "${var.prefix}-datacatalog-policy-tags"
+  project_id = module.common-project.project_id
```
Comment (ludoo) [inline examples/data-solutions/data-platform-foundations/06-common.tf:66]: project id first after source and count ;)

File: examples/data-solutions/data-platform-foundations/variables.tf [hunk 3]
```diff
@@ -33,6 +33,17 @@ variable "composer_config" {
...
+  description = "List of Data Catalog Policy tags to be created with optional IAM binging configuration in {tag => {ROLE => [MEMBERS]}} format."
+  type        = map(map(list(string)))
+  nullable    = false
+  default = {
+    "3_Confidential" = null
+    "2_Private"      = null
+    "1_Sensitive"    = null
+  }
+}
+
```
Comment (ludoo) [inline examples/data-solutions/data-platform-foundations/variables.tf:46]: You can set `nullable = false`, `default = {}` and then ignore the `count` in the module.

**Review by ludoo — APPROVED** [review]
A couple of nits, merge once you take care of those!

File: examples/data-solutions/data-platform-foundations/variables.tf (hunk 3 above)
Comment (lcaggio) [reply examples/data-solutions/data-platform-foundations/variables.tf:46]: good point. Done.

File: examples/data-solutions/data-platform-foundations/06-common.tf (hunk 2 above)
Comment (lcaggio) [reply examples/data-solutions/data-platform-foundations/06-common.tf:64]: Done.

File: examples/data-solutions/data-platform-foundations/06-common.tf (hunk 1 above)
Comment (lcaggio) [reply examples/data-solutions/data-platform-foundations/06-common.tf:63]: Done

**Pull request merged**